File: Scala/Fibonacci/Fibonacci.scala
import scala.annotation.tailrec

/**
 * Calculating a Fibonacci sequence recursively using Scala.
 */
 
object Fibonacci {

    def fib(number: Int) : Int  = {
        if (number == 0) {
            0
        }
        else if(number == 1 ) {
            1
        } else {
           fib(number-1) + fib(number-2)
        }
    }

  def main(args: Array[String]): Unit = {
    for (i <- 1 until 10){
      println(fib(i))
    }

  }
}

File: algorithms/dynamic-programming/fibonacci/go/fibonacci.go
package main

import "fmt"

func fib(n int) int {
	var sequence = make([]int, n + 1, n + 1)
	sequence[0], sequence[1] = 0, 1

	for i := 2; i <= n; i += 1 {
		sequence[i] = sequence[i-1] + sequence[i-2]
	}

	return sequence[n]
}

func main() {
	fmt.Println(fib(10))
}

func fibonacci(n int) int {
	if n <= 0 {
		return 0
	}
	if n == 1 {
		return 1
	}
	prev := 0
	curr := 1
	for i := 2; i <= n; i++ {
		prev, curr = curr, prev+curr
	}
	return curr
}
